db/db_test.cc

```diff
@@ -6677,6 +6677,49 @@ TEST_F(DBTest, CreationTimeOfOldestFile) {
   ROCKSDB_NAMESPACE::SyncPoint::GetInstance()->DisableProcessing();
 }
 
+TEST_F(DBTest, MemoryUsageWithMaxWriteBufferSizeToMaintain) {
+  Options options = CurrentOptions();
+  options.max_write_buffer_size_to_maintain = 1000000;
+  options.write_buffer_size = 16000000;
+  Reopen(options);
+  Random rnd(301);
+  bool memory_limit_increasing = false;
+  uint64_t size_all_mem_table = 0;
+  uint64_t cur_active_mem = 0;
+
+  for (int i = 0; i < 5000; i++) {
+    std::string value = rnd.RandomString(100000);
```

Comment:
I'm wondering if we could scale this value size down (as well as `max_write_buffer_size_to_maintain` and `write_buffer_size`) by say, a factor of 1000 to make the test more efficient?

Reply:
Scaled ```max_write_buffer_size_to_maintain```, ```write_buffer_size``` down by a factor of 100 (wasn't able to reproduce if reduce by 1000) and reduced the  total iteration to 1000.